### ISSUE TITLE
Many errors in log when moving the mouse in Internet explorer

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#10736]
### Steps to reproduce:
* Open IE
* Move the mouse on the page over an edit field (or maybe other form elements) To do this, you can go with virtual cursor on an edit field, e.g. with 'E' letter and then press NVDA+PavNum/ to move the mouse cursor to this edit field.

### Actual behavior:
Many error sounds are heard and many errors can be seen in the log

<details><summary>fragment of log when I am moving the mouse</summary>
<p>

```
WARNING - eventHandler._EventExecuter.next (16:41:44.455) - MainThread (14764):
Could not execute function event_mouseMove defined in NVDAObjects module; kwargs: {'x': 736, 'y': 98}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
ERROR - eventHandler.executeEvent (16:41:44.455) - MainThread (14764):
error executing event: mouseMove on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionMSHTMLIAccessible object at 0x07BD5850> with extra args of {'x': 736, 'y': 98}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 107, in next
  File "extensionPoints\util.pyc", line 170, in callWithSupportedKwargs
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
WARNING - eventHandler._EventExecuter.next (16:41:44.539) - MainThread (14764):
Could not execute function event_mouseMove defined in NVDAObjects module; kwargs: {'x': 736, 'y': 100}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
ERROR - eventHandler.executeEvent (16:41:44.539) - MainThread (14764):
error executing event: mouseMove on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionMSHTMLIAccessible object at 0x07BD5850> with extra args of {'x': 736, 'y': 100}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 107, in next
  File "extensionPoints\util.pyc", line 170, in callWithSupportedKwargs
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
WARNING - eventHandler._EventExecuter.next (16:41:44.779) - MainThread (14764):
Could not execute function event_mouseMove defined in NVDAObjects module; kwargs: {'x': 736, 'y': 102}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
ERROR - eventHandler.executeEvent (16:41:44.789) - MainThread (14764):
error executing event: mouseMove on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionMSHTMLIAccessible object at 0x07BD5850> with extra args of {'x': 736, 'y': 102}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 107, in next
  File "extensionPoints\util.pyc", line 170, in callWithSupportedKwargs
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
WARNING - eventHandler._EventExecuter.next (16:41:44.879) - MainThread (14764):
Could not execute function event_mouseMove defined in NVDAObjects module; kwargs: {'x': 736, 'y': 105}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
ERROR - eventHandler.executeEvent (16:41:44.879) - MainThread (14764):
error executing event: mouseMove on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionMSHTMLIAccessible object at 0x07BD5850> with extra args of {'x': 736, 'y': 105}
Traceback (most recent call last):
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 107, in next
  File "extensionPoints\util.pyc", line 170, in callWithSupportedKwargs
  File "NVDAObjects\__init__.pyc", line 1063, in event_mouseMove
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 427, in makeTextInfo
  File "documentBase.pyc", line 24, in makeTextInfo
  File "NVDAObjects\IAccessible\MSHTML.pyc", line 333, in __init__
TypeError: not all arguments converted during string formatting
IO - inputCore.executeGesture (16:41:46.821) - winInputHook (4484):
Input: kb(desktop):NVDA+f1
```

</p>
</details>

### Expected behavior:
No error

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA 2019.3beta3

#### Windows version:
Win 10

#### Name and version of other software in use when reproducing the issue:
IE 11

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA 2019.2.1 portable: no error

### Note
This issue was originally reported in a French mailing list in IE, Windows Live mail and also sometimes in Edge. I have only checked in IE; I cannot guarantee that problems in WLM and Ege are the same.

**Comment by tapper82** [comment]
Internet explorer is this 2005? Use Edge or even better Firefox.

**Comment by Brian1Gaff** [comment]
Yes if you use IE now funny things can happenwhch did not happen in thhe 
last beta, fortunately it does not affect firefox. Could this be due to the 
use of ie code again rather edge code for some page elements  as I read a 
while back? Maybe these routines need to be made python 3 ready but as they 
were not used nobody knew. Just a suggestion.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
-----

**Comment by Brian1Gaff** [comment]
Yes that as maybe but it used to be fine, some people use ie for other 
reasons.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
-----

**Comment by ehollig** [comment]
@CyrilleB79 how do these errors impact the user? Is there any issues when using NVDA 2019.2? You may hear error tones on the beta, but these could be harmless

**Comment by Adriani90** [comment]
I think IE11 should be still supported due to its popularity in corporate environments. At least until 2025, many big web app vendors like SAP will still stay to the old apps which are working only in IE.
SAP explained that they will move to a cloud based app environment beginning of 2025 and the support for the old app environment (including net viewer and SAP GUI) will be droped. However, until 2025 it is still a long time to go and people are seeing more and more issues with IE and NVDA.

**Comment by Brian1Gaff** [comment]
As an aside does that mean that any software which relied  on the old mshtml 
code will also stop working at that time then?
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Comment by CyrilleB79** [comment]
@tapper82:
I usually use Firefox, but my company force the use of IE for internal pages and app that were designed for this browser.

@ehollig:
I have open this issue because there are a lot of errors in the log and because that was not the case in 2019.2.1. And also because it was mentionned on the NVDA scripts French mailing list. The
goal is to fix a potential regression in NVDA 2019.3.
However, this does not seem to be a critical bug. And I have not seen an impact on the user side. I cannot be sure however that NVDA reads all chunks of text while reading the mouse.

**Comment by agashlin** [comment]
This also seems to be causing issues with reading the names of buttons (and possibly other elements) on mouse move, in IE8. I have a fix.

`event_mouseMove()` [tries to pass a Point into `makeTextInfo()`](https://github.com/nvaccess/nvda/blob/16ec967e3dcadb77d79862fd6dcb3c4868247260/source/NVDAObjects/__init__.py#L1063), assuming that it will hit a `NotImplementedError` exception if the info object can't be created using a Point. It will then retry with POSITION_FIRST. `MSHTMLTextInfo` doesn't accept a point, but when it [tries to format the position](https://github.com/nvaccess/nvda/blob/16ec967e3dcadb77d79862fd6dcb3c4868247260/source/NVDAObjects/IAccessible/MSHTML.py#L333) for the exception, it hits a `TypeError` instead, which throws clear out of `event_mouseMove()`.

I think this was caused by switching to `locationServices.Point` in #9603. This Point is a `collections.namedtuple`, which is also a tuple, while `textInfos.Point` was not. Using a tuple with the `%` operator means that it will attempt to format every element of the tuple, that's not what is intended here.

I'll submit a pull request that wraps the position in a 1-ary tuple first, that fixes the issue for me. There may be other similar issues but I haven't had time to track them down.
